### DISCUSSION TITLE
How to follow via API or CLI?

### DISCUSSION THREAD

**Discussion by benjiwheeler** [bluesky-social/atproto#1707] [How to do it?] [answered]
I'm interested in supporting users in following lists of other users. Say, for example, I have a blog post about science in social media, and I mention 50 Bluesky accounts to follow; I would love to be able to say "here's a tool I made that I've provided these 50 handles, you just have to authorize Bluesky and then you can click once to follow all 50."

Questions:
1) this isn't currently possible with the Bluesky API, right? I don't see anything about being able to Follow via the API.
2) is this something that is on the roadmap, or that there is support for doing? Or, is it something that the AT project has decided not to do?
3) this could obviously also be used to import users collected from other sources -- say, users who you already follow on other platforms. Is that discouraged? Or is it something where the attitude is, this is an agnostic tool, it's not our problem if users use their following lists from elsewhere to identify users here who they want to reconnect to?

**Comment by snarfed** [comment, marked as answer]
You can follow by calling [`com.atproto.repo.createRecord`](https://github.com/bluesky-social/atproto/blob/main/lexicons/com/atproto/repo/createRecord.json) with a [`app.bsky.graph.follow`](https://github.com/bluesky-social/atproto/blob/main/lexicons/app/bsky/graph/follow.json).

**Reply by bnewbold** [reply]
You could also use `com.atproto.repo.applyWrites` to create multiple records at once. A bit more efficient, though a more complex API call. Up to a couple dozen records per write works fine, especially as "follows" are pretty small records.

**Comment by bnewbold** [comment]
FYI, there is a good chance we'll have "lists" in the bsky app soon in a more general form than the current "mute lists". This could be used to share recommended lists of accounts to follow (though may not include the "auto-follow all of these" functionality).

**Comment by StartAutomating** [comment]
[Here's a PowerShell version of .Follow()](https://github.com/StartAutomating/PSA/blob/main/Types/app/bsky/actor/getProfile/Follow.ps1)

The core of it is something like this:

~~~PowerShell
Add-AtprotoRepoRecord -Repo $myBlueSkySession.did -Collection 'app.bsky.graph.follow' -Record ([PSCustomObject][Ordered]@{
    '$type' = 'app.bsky.graph.follow'
    createdAt = [DateTime]::Now.ToString('o')
    subject=$this.did
})
~~~

In this code `$myBlueSkySession.did` is the currently logged on user and `$this` is a profile or actor object (which has a `.did`) property
